foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_8_0/AiopsReferencePDPPythonMigration.java

```diff
@@ -0,0 +1,122 @@
+package com.boozallen.aissemble.upgrade.migration.v1_8_0;
+
+/*-
+ * #%L
+ * aiSSEMBLE::Foundation::Upgrade
+ * %%
+ * Copyright (C) 2021 Booz Allen
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import com.boozallen.aissemble.upgrade.migration.AbstractAissembleMigration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Map;
+
+import static com.boozallen.aissemble.upgrade.util.FileUtils.replaceLiteralInFile;
+
+/**
+ Baton migration class to verify whether a migration to the Policy Decision Point (Python) module files is required from
+ the aiops reference refactor.
+ **/
+public class AiopsReferencePDPPythonMigration extends AbstractAissembleMigration {
+
+    public static final Logger logger = LoggerFactory.getLogger(AiopsReferencePDPPythonMigration.class);
+    public static final Map<String, String> AIOPS_REFERENCE_PYTHON_PROPERTIES_FILE_PACKAGE_MAP = Map.of(
+            "policy-decision-point.aiops-security.properties.vm", "policy-decision-point.aissemble-security.properties.vm",
+            "aiops-security.properties", "aissemble-security.properties",
+            "aiops.authority","aissemble.authority",
+            "aiopskey","aissemblekey",
+            "/deployments/aiops-secure.jks","/deployments/aissemble-secure.jks"
+            );
+
+    /**
+     * Function to check whether the migration is necessary.
+     * @param file file to check
+     * @return shouldExecute - whether the migration is necessary.
+     */
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
+        boolean shouldExecute = false;
+
+        try {
+            if(file.getName().equals("aiops-security.properties")) {
+                shouldExecute = true;
+            } else {
+                BufferedReader aiopsReferencePythonDockerfileConfig = new BufferedReader((new FileReader(file)));
+
+                String line;
+                while((line = aiopsReferencePythonDockerfileConfig.readLine()) !=null && !shouldExecute) {
+                    for (String key : AIOPS_REFERENCE_PYTHON_PROPERTIES_FILE_PACKAGE_MAP.keySet()) {
+                        if (line.contains(key)) {
+                            shouldExecute = true;
+                        }
+                    }
+                }
+            }
+        } catch (IOException e) {
+            logger.error("Error in determining whether a 'AIOPS' renamed properties file requires a migration.");
+        }
+
+        return shouldExecute;
+    }
+
+    /**
+     * Performs the migration if the shouldExecuteOnFile() returns true.
```

Comment:
minor: shouldExecuteOnFile() is external to this method.  Can the description instead just say that it renames a file name?  Meaning, technically this method can be called by any other private method regardless of what shouldExecuteOnFile() returns

Reply:
Updated the description to better describe functionality.